sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -25,17 +25,24 @@ type globalEndpointManager struct {
 }
 
 // NewGlobalEndpointManager creates a new global endpoint manager.
-func NewGlobalEndpointManager(client *Client, preferredRegions []string) (*globalEndpointManager, error) {
+func newGlobalEndpointManager(client *Client, preferredRegions []string, refreshInterval time.Duration) (*globalEndpointManager, error) {
 	endpoint, err := url.Parse(client.endpoint)
 	if err != nil {
 		return nil, err
 	}
-
-	return &globalEndpointManager{
+	gem, err := &globalEndpointManager{
 		client:           client,
 		preferredRegions: preferredRegions,
 		locationCache:    newLocationCache(preferredRegions, *endpoint),
 	}, nil
+	if err != nil {
+		return nil, err
+	}
+	gem.stopBackgroundRefresh()
+	gem.startBackgroundRefresh(refreshInterval)
+	defer gem.stopBackgroundRefresh()
```

Comment:
Won't this `defer` stop the background refresh when the constructor exists/returns?